jgiven-core/src/main/java/com/tngtech/jgiven/impl/params/DefaultAsProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.tngtech.jgiven.impl.params;
+
+import java.lang.reflect.Method;
+
+import com.tngtech.jgiven.annotation.As;
+import com.tngtech.jgiven.annotation.AsProvider;
+import com.tngtech.jgiven.config.AbstractJGivenConfiguration;
+import com.tngtech.jgiven.config.ConfigurationUtil;
+import com.tngtech.jgiven.impl.util.WordUtil;
+
+/**
+ * The default provider for a stage method, scenario or scenario class.
+ *
+ */
+public class DefaultAsProvider implements AsProvider {
+
+    /**
+     * Returns the value of the {@link As} annotation if a value was specified.
+     * Otherwise, this transforms the method name into a readable sentence and returns it.
+     */
+    @Override
+    public String as( As annotation, Method method ) {
+        if( annotation != null && annotationHasExplicitValue( annotation ) ) {
+            return annotation.value();
+        }
+
+        String methodName = method.getName();
```

Comment:
So I think one can make either argument for or against having this in the DefaultAsProvider. My rationale is that this way, a user can extend from this provider and utilize this, e.g., in the spirit of the original motivation for #189:

````
@As( provider = ReplaceTodayProvider.class )
public SELF a_request_with_request_date_TODAY() { … }

public static class ReplaceTodayProvider extends DefaultAsProvider {
    @Override
    public String as( As annotation, Method method ) {
        String words = super.as( annotation, method );
        return String.replace( "TODAY", formatDate( new Date() ) );
    }
}
```